airbyte-cdk/python/airbyte_cdk/sources/concurrent_source/concurrent_read_processor.py

```diff
@@ -183,7 +189,15 @@ def is_done(self) -> bool:
         2. There are no more streams to read from
         3. All partitions for all streams are closed
         """
-        return all([self._is_stream_done(stream_name) for stream_name in self._stream_name_to_instance.keys()])
+        is_done = all([self._is_stream_done(stream_name) for stream_name in self._stream_name_to_instance.keys()])
+        if is_done and self._exceptions_per_stream_name:
+            error_message = self._generate_failed_streams_error_message(self._exceptions_per_stream_name)
+            self._logger.info(error_message)
+            # We still raise at least one exception when a stream raises an exception because the platform currently relies
+            # on a non-zero exit code to determine if a sync attempt has failed. We also raise the exception as a config_error
+            # type because this combined error isn't actionable, but rather the previously emitted individual errors.
+            raise AirbyteTracedException(message=error_message, internal_message="Concurrent read failure", failure_type=FailureType.config_error)
```

Comment:
Having a internal_message is different than abstract_source. I had to add an internal message as the scenario framework will assert on the internal_message. Leaving this empty would generate `'NoneType' object is not subscriptable`